File: 백준/Silver/14888. 연산자 끼워넣기/연산자 끼워넣기.cc
#include <iostream>
using namespace std;

int n, maxResult = -1000000000, minResult = 1000000000;
int operands[11], operators[4];

void backtracking(int result, int index) {
    if(index == n) {
        if(result > maxResult) maxResult = result;
        if(result < minResult) minResult = result;
        return;
    } else {
        for(int i = 0; i < 4; i++) {
            if(operators[i] > 0) {
                operators[i]--;
                if(i == 0) backtracking(result + operands[index], index + 1);
                else if(i == 1) backtracking(result - operands[index], index + 1);
                else if(i == 2) backtracking(result * operands[index], index + 1);
                else if(i == 3) backtracking(result / operands[index], index + 1);
                operators[i]++;
            }
        }
    }
}

int main() {
    cin >> n;
    
    for(int i = 0; i < n; i++) cin >> operands[i];
    for(int i = 0; i < 4; i++) cin >> operators[i];
    
    backtracking(operands[0], 1);
    
    cout << maxResult << '\n' << minResult;
    
}

File: 백준/Silver/14888. 연산자 끼워넣기/연산자 끼워넣기.swift
import Foundation

let n = Int(readLine()!)!
var numbers = readLine()!.split(separator: " ").map { Int(String($0))! }
var symbols = readLine()!.split(separator: " ").map { Int(String($0))! } // + - x /

var max = Int.min
var min = Int.max

func backtracking(_ result: Int, _ index: Int) {
    if index == n {
        if result > max { max = result }
        if result < min { min = result }
        return
    } else {
        for i in 0..<4 {
            if symbols[i] > 0 {
                symbols[i] -= 1
                
                switch i {
                    case 0:
                        backtracking(result + numbers[index], index + 1)
                    case 1:
                        backtracking(result - numbers[index], index + 1)
                    case 2:
                        backtracking(result * numbers[index], index + 1)
                    case 3:
                        backtracking(result / numbers[index], index + 1)
                    default:
                        fatalError("default Error")
                }
                
                symbols[i] += 1
            }
        }
    }
}

func solution() {
    backtracking(numbers[0], 1)
    
    print("\(max)\n\(min)")
}

solution()